fix for https://github.com/wso2/docs-apim/issues/1129 - master (#1231)
wso2/docs-apim: en/docs/learn/analytics/configuring-keystores-in-apim-analytics.md

               ... 
         ```
        The elements in the above configuration are described below:
          
        | Element                | Description                                                      |
        |------------------------|------------------------------------------------------------------|
        | **privateKeyAlias**    | Private key alias                                                |
        | **keyStoreLocation**   | Name of the keystore file. **The file must be located in the `<API-M_ANALYTICS_HOME>/resources/security` directory.** |

    2. Provide the keystore and private key password in the Base64 encoded format in the `<ANALYTICS_HOME>/conf/dashboard/master-keys.yaml` file.

        ??? info "Click here for more info on the `master-keys.yaml` file"
            The base64 encoded plaintext keystore password and the base64 encoded plaintext private password of the keystore used for Secure Vault is defined in the master-keys.yaml file. WSO2 Carbon Server reads this file to obtain these two passwords, so that it can decrypt the passwords in the `secrets.properties` file in order to start the server.

> 4. Set the correct keystore password.

Reviewer: Let's add the configuration of the Keystore name in here as well.
```suggestion
4. Set the correct Keystore name.
    If you have created a new Keystore, then define the correct path for the new Keystore in the `-Djavax.net.ssl.keyStore` system property in the `<ANALYTICS_HOME>/wso2/dashboard/bin/carbon.sh` file when you change the Keystore.
```
After this, we can add the changing password step. Please do the same for the worker instructions(in this file) as well. Also please do the same change for the 3.1.0 version and 3.0.0 versions as well.

Author: Fixed in https://github.com/wso2/docs-apim/pull/1231/commits/f872500a49ef0c6223fab16659a66f2e133faa22

## Configuring keystores in worker profile

!!! note
    Other than explicitly mentioned, you need to do all the following configuration updates related to the worker profile in the `<API-M_ANALYTICS_HOME>/conf/worker/deployment.yaml` file.

1. Update the databridge configurations.

    This is used for authenticating communication over SSL/TLS between API Manager and API-M Analytics. Handles thrift and binary transports.

    ```yaml
    databridge.config:
      ... 
      keyStoreLocation : ${sys:carbon.home}/resources/security/wso2carbon.jks
      keyStorePassword : wso2carbon
      ... 
    ```
    The elements in the above configuration are described below:
   
    | Element                | Description                                                      |
    |------------------------|------------------------------------------------------------------|
    | **keyStoreLocation**   | Name of the keystore file. **The file must be located in the `<API-M_ANALYTICS_HOME>/resources/security` directory.** |
    | **keyStorePassword**   | Keystore password                                                |

2. Update the HTTP transport configurations.

>     This is used for authentication, encryption, and signing for the analytics dashboard web application.

Reviewer: ```suggestion
    This is used for authentication, encryption, and signing for the analytics worker.
```

Author: Fixed in https://github.com/wso2/docs-apim/pull/1231/commits/f872500a49ef0c6223fab16659a66f2e133faa22

          The elements in the above configuration are described below:
            
          | Element                | Description                                                      |
          |------------------------|------------------------------------------------------------------|
          | **privateKeyAlias**    | Private key alias                                                |
          | **keyStoreLocation**   | Name of the keystore file. **The file must be located in the `<API-M_ANALYTICS_HOME>/resources/security` directory.** |

     2. Provide the keystore and private key password in the Base64 encoded format in the `<ANALYTICS_HOME>/conf/worker/master-keys.yaml` file.

        ??? info "Click here for more info on the `master-keys.yaml` file"
            The base64 encoded plaintext keystore password and the base64 encoded plaintext private password of the keystore used for Secure Vault is defined in the master-keys.yaml file. WSO2 Carbon Server reads this file to obtain these two passwords, so that it can decrypt the passwords in the `secrets.properties` file in order to start the server.

4. Set the new keystore details when you change the keystore.

>       1. Navigate to the <ANALYTICS_HOME>/wso2/dashboard/bin/carbon.sh` file.

Reviewer: ```suggestion
      1. Navigate to the <ANALYTICS_HOME>/wso2/worker/bin/carbon.sh` file.
```

Author: Fixed in https://github.com/wso2/docs-apim/pull/1231/commits/f872500a49ef0c6223fab16659a66f2e133faa22

Reviewer: Still, this change is not fixed.
